.github/workflows/release.yml

```diff
@@ -57,20 +82,93 @@ jobs:
           tail +$begin ./CHANGELOG.md | head -$end >> $GITHUB_ENV
           echo 'EOF' >> $GITHUB_ENV
       - name: Create release
-        if: ${{ steps.push.outcome == 'success' && steps.message.outcome == 'success' }}
+        if: steps.push.outcome == 'success' && steps.message.outcome == 'success' && github.event.inputs.test_release == 'false'
         uses: ncipollo/release-action@v1
         with:
           body: ${{ env.MESSAGE }}
           token: ${{ secrets.COVALENT_OPS_BOT_TOKEN }}
           tag: ${{ env.RELEASE }}
-          prerelease: true
+          prerelease: ${{ github.event.inputs.prerelease }}
+      - name: Alert Slack
+        if: failure()
+        uses: rtCamp/action-slack-notify@v2
+        env:
+          SLACK_CHANNEL: "covalent"
+          SLACK_USERNAME: "CovalentOpsBot"
+          SLACK_MESSAGE: "The release.yml workflow is failing in ${{ github.ref }}!"
+          SLACK_COLOR: ${{ job.status }}
+          SLACK_TITLE: ":warning:  Attention Required  :warning:"
+          SLACK_WEBHOOK: ${{ secrets.SLACK_ALERT_WEBHOOK }}
+  pypi:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out release tag
+        uses: actions/checkout@v2
+        if: github.event.inputs.prerelease == 'false' && github.event.inputs.stable_version
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+          ref: "v${{ github.event.inputs.stable_version }}"
+      - name: Check out master
+        uses: actions/checkout@v2
+        if: github.event.inputs.prerelease == 'true'
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install twine
+      - name: Set up Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+      - name: Build Webapp
+        run: |
+          cd ./covalent_ui/webapp
+          yarn install
+          yarn build
+      - name: Build Prerelease Distribution
+        if: github.event.inputs.prerelease == 'true'
+        run: python setup.py egg_info --tag-build=pre sdist
+      - name: Build Stable Distribution
+        if: github.event.inputs.prerelease == 'false' && github.event.inputs.stable_version
+        run: python setup.py sdist
+      - name: Validate Distribution
+        run: |
+          if [ ${{ github.event.inputs.prerelease }} = 'false' ] && \
+             [ -z ${{ github.event.inputs.stable_version }} ] ; then
+            echo "You can't create a stable release without specifying the stable version number."
+            exit 1
+          fi
+          VERSION="$(cat ./VERSION)"
+          cd dist
+          tar xzf cova-${VERSION}rc0.tar.gz
+          diff -x tests -r cova-${VERSION}rc0/covalent ../covalent
```

Comment:
the name of the tarball will depend on whether it's a pre-release or stable release

Reply:
oh right of course, thanks for catching that